storage/http_client.go

```diff
@@ -423,6 +423,28 @@ func (c *httpStorageClient) ListBucketACLs(ctx context.Context, bucket string, o
 	return toBucketACLRules(acls.Items), nil
 }
 
+func (c *httpStorageClient) UpdateBucketACL(ctx context.Context, bucket string, entity ACLEntity, role ACLRole, opts ...storageOption) (*ACLRule, error) {
+	s := callSettings(c.settings, opts...)
+	acl := &raw.BucketAccessControl{
+		Bucket: bucket,
+		Entity: string(entity),
+		Role:   string(role),
+	}
+	var aclRule ACLRule
+	var err error
+	err = run(ctx, func() error {
+		req := c.raw.BucketAccessControls.Update(bucket, string(entity), acl)
+		configureACLCall(ctx, s.userProject, req)
+		acl, err = req.Do()
+		aclRule = toBucketACLRule(acl)
+		return err
+	}, s.retry, true)
```

Comment:
Idempotency should be taken from s (and will be false ultimately)

Reply:
Right, it's not idempotent. Fixed.

Reply:
As I mentioned, you should use the setting in s (`s.idempotent`)

Reply:
Updated to use `s.idempotent` 🙈